pkg/utils/netlink.go

```diff
@@ -73,148 +80,261 @@ type NetlinkWrapper struct {
 
 // NewNetlinkWrapper creates initialized instance of NetlinkWrapper
 func NewNetlinkWrapper() *NetlinkWrapper {
-	return &NetlinkWrapper{tracer: otel.Tracer("")}
+	return NewNetlinkWrapperWithArgs(true)
+}
+
+// NewNetlinkWrapperWithArgs creates initialized instance of NetlinkWrapper
+// based on passing arguments
+func NewNetlinkWrapperWithArgs(enableTracer bool) *NetlinkWrapper {
+	netlinkWrapper := &NetlinkWrapper{}
+	if enableTracer {
+		netlinkWrapper.tracer = otel.Tracer("")
```

Comment:
I see that we need to check a tracer all the time here and there...
Is it possible to use [NoopTracerProvider](https://pkg.go.dev/go.opentelemetry.io/otel/api/trace#NoopTracerProvider) here, so we use that noop object and not checking for nil all the time?

Reply:
That is a very good point. Let me take a look